waku/waku_rln_relay/rln/wrappers.nim

```diff
@@ -163,7 +163,8 @@ proc poseidon*(data: seq[seq[byte]]): RlnRelayResult[array[32, byte]] =
 when defined(rln_v2):
   proc toLeaf*(rateCommitment: RateCommitment): RlnRelayResult[seq[byte]] =
     let idCommitment = rateCommitment.idCommitment
-    let userMessageLimit =  cast[array[32, byte]](rateCommitment.userMessageLimit)
+    var userMessageLimit: array[32, byte]
+    discard userMessageLimit.copyFrom(toBytes(rateCommitment.userMessageLimit, Endianness.littleEndian))
```

Comment:
Given that `toLeaf` returns a `Result` I wonder if we could handle the possible `copyFrom` error at this point.

Reply:
addressed in d3864182